deploy-cluster-aws/launch_ec2_nodes.py

```diff
@@ -36,14 +38,52 @@
 parser.add_argument("--tag",
                     help="tag to add to all launched instances on AWS",
                     required=True)
-parser.add_argument("--nodes",
-                    help="number of nodes in the cluster",
-                    required=True,
-                    type=int)
+parser.add_argument("--deploy-conf-file",
+                    help="AWS deployment configuration file",
+                    required=True)
 args = parser.parse_args()
-
 tag = args.tag
-num_nodes = int(args.nodes)
+deploy_conf_file = args.deploy_conf_file
+
+# Import all the variables set in the AWS deployment configuration file
+# (Remove the '.py' from the end of deploy_conf_file.)
+cf = importlib.import_module(deploy_conf_file[:-3])
+try:
+    NUM_NODES = cf.NUM_NODES
+    BRANCH = cf.BRANCH
+    WHAT_TO_DEPLOY = cf.WHAT_TO_DEPLOY
+    USE_KEYPAIRS_FILE = cf.USE_KEYPAIRS_FILE
+    IMAGE_ID = cf.IMAGE_ID
+    INSTANCE_TYPE = cf.INSTANCE_TYPE
+except AttributeError as e:
+    print('One of the AWS deployment configuration settings was '
+          'not set in the AWS deployment configuration file ' +
+          '{}'.format(deploy_conf_file))
+    print('Read this traceback to find out which one (in ALL_CAPS):')
+    raise
```

Comment:
what about using multiple `try/except` clauses, to pinpoint the exact problem, e.g.:

```python
err_msg = ('The AWS deployment configuration settings "{}" was not set in the '                                                                                                                                                                                                            
           'AWS ...')                                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                           
try:                                                                                                                                                                                                                                                                                       
    NUM_NODES = cf.NUM_NODES                                                                                                                                                                                                                                                               
except AttributeError:                                                                                                                                                                                                                                                                     
    raise ImproperlyConfigured(err_msg.format('NUM_NODES'))                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                           
try:                                                                                                                                                                                                                                                                                       
    BRANCH = cf.BRANCH                                                                                                                                                                                                                                                                     
except AttributeError:                                                                                                                                                                                                                                                                     
    raise ImproperlyConfigured(err_msg.format('BRANCH'))                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                           
try:                                                                                                                                                                                                                                                                                       
    WHAT_TO_DEPLOY = cf.WHAT_TO_DEPLOY                                                                                                                                                                                                                                                     
except AttributeError:                                                                                                                                                                                                                                                                     
    raise ImproperlyConfigured(err_msg.format('WHAT_TO_DEPLOY'))                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                           
try:                                                                                                                                                                                                                                                                                       
    USE_KEYPAIRS_FILE = cf.USE_KEYPAIRS_FILE                                                                                                                                                                                                                                               
except AttributeError:                                                                                                                                                                                                                                                                     
    raise ImproperlyConfigured(err_msg.format('USE_KEYPAIRS_FILE'))                                                                                                                                                                                                                        
                                                                                                                                                                                                                                                                                           
try:                                                                                                                                                                                                                                                                                       
    IMAGE_ID = cf.IMAGE_ID                                                                                                                                                                                                                                                                 
except AttributeError:                                                                                                                                                                                                                                                                     
    raise ImproperlyConfigured(err_msg.format('IMAGE_ID'))                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                           
try:                                                                                                                                                                                                                                                                                       
    INSTANCE_TYPE = cf.INSTANCE_TYPE                                                                                                                                                                                                                                                       
except AttributeError:                                                                                                                                                                                                                                                                     
    raise ImproperlyConfigured(err_msg.format('INSTANCE_TYPE'))                                                                                         ```